avm/res/azure-stack-hci/cluster/arc-gateway/main.bicep

```diff
@@ -0,0 +1,40 @@
+metadata name = 'Arc Gateway'
+metadata description = 'This module deploys an Arc Gateway.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Arc Gateway to deploy.')
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Arc Gateway allowed features.')
+param allowedFeatures array = ['*']
+
+@description('Optional. Arc Gateway type. Detaults to Public.')
+@allowed(['Public'])
+param gatewayType string = 'Public'
+
+resource arcGateway 'Microsoft.HybridCompute/gateways@2024-05-20-preview' = {
```

Comment:
This should be it's own module as it's not a child, but a top level resource. Further, unless a HybridCompute/gateway resource is not usually always deployed with a azure-stack-hci/custer, their joint deployed should likely be implemented in a pattern module, not a resource module.

For example: Ideally we should have one `res` module for each resource type you want to deploy. In this case that could be
- `res/azure-stack-hci/cluster`
- `res/hybrid-compute/gateway`
- ...
Which themselves would be published to the public registry. A pattern module could then reference those individual modules to assemble a workload.